op-guide/gc.md

```diff
@@ -0,0 +1,77 @@
+```
+
+The duration strings are a sequence of a number with the time unit, such as `24h`, `2h30m` and `2.5h`. The time units you can use include "h", "m" and "s".
+
+> **Note**: When you set `tikv_gc_life_time` to a large number (like days or even months) in a data updated frequently scenario, some problems as follows may occur: 
+
+    - The more versions of the data, the more disk storage space is occupied.
+    - A large number of history versions might slow down the query. They may affect range queries like `select count(*) from t`.
+    - If `tikv_gc_life_time` is suddenly turned to a smaller value during operation, a great deal of old data may be deleted in a short time, causing I/O pressure.
+
+`tikv_gc_concurrency` indicates the GC concurrency. It is set to `1` by default. In this case, a single thread operates and threads send request to each Region and wait for the response one by one. You can set the variable value larger to improve the system performance, but keep the value smaller than 128. 
+
+`tikv_gc_leader_uuid`, `tikv_gc_leader_desc`, `tikv_gc_leader_lease` indicate the current GC leader information. `tikv_gc_last_run_time` indicates the last time GC works.
+
+`tikv_gc_safe_point` indicates the time, versions before which are cleared by GC and versions after which are readable.
+
+## Implementation details
+
+The GC implementation process is complex. Clearing the data that is not used any more should be on the premise that data consistency is guaranteed. The process of doing GC is as below:
+
+### 1. Resolve locks
+
+The TiDB transaction is based on Google Percolator. The transaction committing is a two-phase committing process. When the first phase is finished, all the related keys are locked. Among these locks, one is the primary lock and the others are secondary locks which contain a pointer of the primary locks; in the secondary phase, the key with the primary lock gets a write record and its lock is removed. The write record indicates the write or delete operation in the history or the transactional rollback record of this key. Replacing the primary lock by which write record indicates whether the corresponding transaction is committed successfully. Then all the secondary locks are replaced successively. If the threads to replace the secondary locks fail, these locks are retained. During GC, locks whose timestamp is before the safe point will be replaced by the corresponding write record based on the transaction committing status.
+
+This step is necessary, because you are not informed whether this transaction is successful if GC has cleared the write record of the primary lock, thus data consistency cannot be guaranteed.
+
+### 2. Delete ranges
+
+The `DeleteRanges` operation is usually necessary after the operation such as `drop table`. It is used to delete a range which may be very large. If the `use_delete_range` option of TiKV is not enabled, TiKV deletes the keys in the range.
```

Comment:
"The `DeleteRanges` operation is usually necessary after the operation such as `drop table`. It is used to delete a range which may be very large." -> "`DeleteRanges` is usually executed after operations like `drop table`, used to delete a range which might be very large."